### PR TITLE
OSASINFRA-3426 -  Adding ShiftStack ingress-perf Periodic CI Job

### PR DESCRIPTION
**Pull request by MiguelCarpio**
Reproduce the OVN Perf Scale CI jobs, implemented by OpenShift QE for their CI environment, to our ShifStack CI environment.

As a telco provider, we need to certify the performance of the data plane network traffic running on OpenShift on OpenStack; this involves the performance test of the Ingress and E-W traffic generated in the OVN layer.

### CODE REVIEW

**Comment by MiguelCarpio** [comment]
> /pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

couldn't get current server API group list: Get "https://api.0w7s4b3c-d2868.shiftstack.devcluster.openshift.com:6443/api?timeout=32s": dial tcp 192.168.25.29:6443: i/o timeout
error: unable to recognize "STDIN": Get "https://api.0w7s4b3c-d2868.shiftstack.devcluster.openshift.com:6443/api?timeout=32s": dial tcp 192.168.25.29:6443: i/o timeout

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
Routing issues. Loading the proxy env is needed to reach the API server.

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.16.yaml [hunk 1]
```diff
@@ -204,6 +204,13 @@ tests:
     - ref: openstack-test-openstack
     workflow: openshift-e2e-openstack-ipi
   timeout: 7h0m0s
+- as: e2e-openstack-ingress-perf
+  minimum_interval: 24h
```
Comment (EmilienM) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.16.yaml:208]: That's too frequent and not necessary.
Maybe once every two weeks is enough.
I don't think that ingress performance is going to change that much.

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.16.yaml [hunk 2]
```diff
@@ -204,6 +204,13 @@ tests:
...
+  steps:
+    cluster_profile: openstack-vh-mecha-az0
+    env:
+      LOG_LEVEL: debug
```
Comment (EmilienM) [inline ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.16.yaml:212]: a comment would be nice to explain why we enable debug log level.

File: ci-operator/step-registry/openshift/e2e/openstack/OWNERS [hunk 3]
```diff
@@ -1,4 +1,4 @@
 approvers:
 - openstack-approvers
 reviewers:
-- openstack-reviewers
+- openstack-reviewers
```
Comment (EmilienM) [inline ci-operator/step-registry/openshift/e2e/openstack/OWNERS:4]: unnecessary change

File: ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+# that libcurl doesn't recognize the uppercase variables).
+if test -f "${SHARED_DIR}/proxy-conf.sh"
+then
+	# shellcheck disable=SC1090
+	source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+# Getting all worker nodes' names
+mapfile -t WORKER_NODES < <(  oc get nodes -o 'jsonpath={range .items[*]}{.metadata.name}{"\n"}{end}' | grep worker )
+
```
Comment (EmilienM) [inline ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh:20]: there is no error management in this script, it'll be nice to have it; so it makes it more robus.
For example return an error if no worker was found.

File: ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+# Adding the app label to the first worker node
+oc label nodes/${WORKER_NODES[0]} node-role.kubernetes.io/app=
+
+# Adding the infra label to the rest of the worker nodes
+for i in "${WORKER_NODES[@]:1}"; do oc label nodes/$i node-role.kubernetes.io/infra=; done
+
```
Comment (EmilienM) [inline ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh:26]: what if we have 1 worker? please handle it or return an error.

File: ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+# Adding the defaultNodeSelector field with the appropriate node selector
+wget https://github.com/mikefarah/yq/releases/latest/download/yq_linux_amd64 -O /tmp/yq && chmod +x /tmp/yq
```
Comment (EmilienM) [inline ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh:28]: no please don't do that; jobs run in containers. If you need yq, you need to find an image with jq in there (`openstack-installer` probably?).

File: ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+SCHEDULER_CLUSTER_YAML=/tmp/scheduler.cluster.yaml
+oc get scheduler cluster -o yaml > ${SCHEDULER_CLUSTER_YAML}
+cat ${SCHEDULER_CLUSTER_YAML} | /tmp/yq '.spec = .spec + {"defaultNodeSelector": "node-role.kubernetes.io/infra=\"\""}' | tee ${SCHEDULER_CLUSTER_YAML}.tmp 
+oc apply -f ${SCHEDULER_CLUSTER_YAML}.tmp
+rm -f ${SCHEDULER_CLUSTER_YAML}*
```
Comment (EmilienM) [inline ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-commands.sh:34]: the last 5 lines can be replaced in a one-liner with
```
oc patch scheduler cluster --type=json -p '[{ "op": "replace", "path" ...
```

File: ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-ref.yaml [hunk 8]
```diff
@@ -0,0 +1,10 @@
+ref:
+  as: openshift-e2e-openstack-perfscale-conf-infralabels
+  from: cli
```
Comment (EmilienM) [inline ci-operator/step-registry/openshift/e2e/openstack/perfscale/conf-infralabels/openshift-e2e-openstack-perfscale-conf-infralabels-ref.yaml:3]: like I said, change the image to get one with `jq`.

File: ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.16.yaml (hunk 2 above)
Comment (MiguelCarpio) [reply ci-operator/config/shiftstack/shiftstack-ci/shiftstack-shiftstack-ci-main__periodic-4.16.yaml:212]: Oh, I forgot to remove it. Done!

File: ci-operator/step-registry/openshift/e2e/openstack/OWNERS (hunk 3 above)
Comment (MiguelCarpio) [reply ci-operator/step-registry/openshift/e2e/openstack/OWNERS:4]: make job thing

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf

**Comment by MiguelCarpio** [comment]
/pj-rehearse periodic-ci-shiftstack-shiftstack-ci-main-periodic-4.16-e2e-openstack-ingress-perf
